src/core/libmaven/databases.cpp

```diff
@@ -20,9 +20,13 @@ int Databases::loadCompoundCSVFile(string filename) {
     while (getline(myfile,line)) {
         //This is used to write commands
         if (!line.empty() && line[0] == '#') continue;
-        
+
         //trim spaces on the left
-        line.erase(line.find_last_not_of(" \n\r\t")+1);
+        size_t found = line.find_last_not_of(" \n\r\t");
```

Comment:
This can be simplied to:
`line.erase(line.find_last_not_of("\r\n\t")+1);`

That's also how it is done in GUI:
https://github.com/ElucidataInc/ElMaven/blob/92b75ba3bb2f43d1bd2a4032150ae3fb149a0162/src/gui/mzroll/database.cpp#L471

Also, for me, on Windows the GUI crashes for the attached DB file but the CLI does not
Can you test this on Windows once ?

[KNOWNS.zip](https://github.com/ElucidataInc/ElMaven/files/2475163/KNOWNS.zip)

Reply:
@rish9511 'found' stores whether there are any other characters in this line apart from " \n\t\r" or not. If this is an empty line, don't send it to csvParser in mzUtils::get_fields() and move to the next line. get_fields is where the app crashes.

I forgot to make this same change for the GUI that uses database.cpp which is probably why GUI is crashing for this file. While I'm at it, I'll also add a test for databases.cpp. Thanks for reminding me!

Reply:
@rish9511 'found' stores whether there are any other characters in this line apart from " \n\t\r" or not. If this is an empty line, don't send it to csvParser in mzUtils::get_fields() and move to the next line. get_fields is where the app crashes.

I forgot to make this same change for the GUI that uses database.cpp which is probably why GUI is crashing for this file. While I'm at it, I'll also add a test for databases.cpp. Thanks for reminding me!

Reply:
@rish9511 I've pushed the required changes. Let me know if this works for you now.

Reply:
👍